src/cli/mod.rs

```diff
@@ -125,48 +125,33 @@ impl Opt {
             no_color: color.is_no_color(),
         };
 
-        let home = match home.or_else(home::home_dir).ok_or_else(|| {
+        let base_dirs = match directories::BaseDirs::new().ok_or_else(|| {
             anyhow!(
-                "failed to determine the current user's home directory, try using the `--home` \
-                 option"
+                "failed to determine the current user's home directory, please set the `HOME` environment variable"
             )
         }) {
-            Ok(home) => home,
+            Ok(base_dirs) => base_dirs,
             Err(err) => {
                 log_error(output.no_color, Color::Red, "error", &err);
                 process::exit(1);
             }
         };
 
-        let xdg_config_user = env::var_os("XDG_CONFIG_HOME").map(PathBuf::from);
-        let xdg_data_user = env::var_os("XDG_DATA_HOME").map(PathBuf::from);
-
-        // Note: `XDG_RUNTIME_DIR` is not checked as it can be set by the system rather
-        // than the user, and cannot be relied upon to indicate a preference for XDG
-        // directory layout.
-        let using_xdg = any!(
-            xdg_data_user,
-            xdg_config_user,
-            env::var_os("XDG_CACHE_HOME"),
-            env::var_os("XDG_DATA_DIRS"),
-            env::var_os("XDG_CONFIG_DIRS")
-        );
-
-        let (config_pre, data_pre) = if using_xdg {
-            (
-                xdg_config_user
-                    .unwrap_or_else(|| home.join(".config"))
-                    .join("sheldon"),
-                xdg_data_user
-                    .unwrap_or_else(|| home.join(".local/share"))
-                    .join("sheldon"),
+        let project_dirs = match directories::ProjectDirs::from("rs", "cli", "sheldon").ok_or_else(|| {
+            anyhow!(
+                "failed to determine the current user's home directory, please set the `HOME` environment variable"
             )
-        } else {
-            (home.join(".sheldon"), home.join(".sheldon"))
+        }) {
+            Ok(project_dirs) => project_dirs,
+            Err(err) => {
+                log_error(output.no_color, Color::Red, "error", &err);
+                process::exit(1);
+            }
         };
 
-        let config_dir = config_dir.unwrap_or(config_pre);
-        let data_dir = data_dir.unwrap_or(data_pre);
+        let home = home.unwrap_or(base_dirs.home_dir().to_path_buf());
+        let config_dir = config_dir.unwrap_or(project_dirs.config_dir().to_path_buf());
+        let data_dir = data_dir.unwrap_or(project_dirs.data_dir().to_path_buf());
```

Comment:
Let's use `.unwrap_or_else` to prevent unnecessary clones in the case that an option is set.

```rust
let home = home.unwrap_or_else(|| base_dirs.home_dir().to_owned());
let config_dir = config_dir.unwrap_or_else(|| project_dirs.config_dir().to_owned());
let data_dir = data_dir.unwrap_or_else(|| project_dirs.data_dir().to_owned());
```